README_python.md

```diff
@@ -170,35 +200,240 @@ essentially zero cost, with storage being shared. Thus, we can usually pass it
+fully defined, in a way that matches the type-and-dimension constraints
+specified.
+
+It is required that the `generate()` method be defined by the Generator.
+
+(Note that, by convention, Halide Generators use `g` instead of `self` in their
+`generate()` method to make the expression language terser; this is not in any
+way required, but is recommended.)
+
+### Using a Generator for JIT compilation
+
+You can use the `compile_to_callable()` method to JIT-compile a Generator into a
+`hl.Callable`, which is (essentially) just a dynamically-created function.
+
+```
+import LogicalOpGenerator
+import imageio
+import numpy as np
+
+# Instantiate a Generator -- we can only set the GeneratorParams
+# by passing in a dict to the Generator's constructor
+or_op_generator = LogicalOpGenerator({"op": "or"})
+
+# Now compile the Generator into a Callable
+or_filter = or_op_generator.compile_to_callable()
+
+# Read in some file for input
+input_buf = imageio.imread("/path/to/some/file.png")
+assert input_buf.dtype == np.uint8
+
+# create a Buffer-compatible object for the output; we'll use np.array
+output_buf = np.empty(input_buf.shape, dtype=input_buf.dtype)
 
-### Using a C++ Generator from Python
+# Note, Python code throws exception for error conditions rather than returning an int
+or_filter(input_buf, 0x7f, output_buf)
 
-Let's say we have a very simple Generator:
+# Note also that we can use named arguments for any/all, in the Python manner:
+or_filter(mask=0x7f, input=input_buf, output=output_buf)
 
+imageio.imsave("/tmp/or.png", output_buf)
 ```
-class MyFilter : public Halide::Generator<Simple> {
-public:
-    Input<Buffer<uint8_t, 2>> input{"input"};
-    Input<uint8_t> mask{"mask"};
-    Output<Buffer<uint8_t, 2>> output{"output"};
-
-    void generate() {
-        output(x, y) = input(x, y) & mask;
-        output.vectorize(x, 8).compute_root();
-    }
-};
-HALIDE_REGISTER_GENERATOR(MyFilter, my_filter)
+
+By default, a Generator will produce code targeted at `Target("host")` (or the
+value of the `HL_JIT_TARGET` environment variable, if set); you can override
+this behavior selectively by activating a `GeneratorContext` when the Generator
+is *created*:
+
 ```
+import LogicalOpGenerator
+
+# Compile with debugging enabled
+t = hl.Target("host-debug")
+with hl.GeneratorContext(t):
+    or_op_generator = LogicalOpGenerator({"op": "or"})
+    or_filter = or_op_generator.compile_to_callable()
+```
+
+### Using a Generator for AOT compilation
 
 If you are using CMake, the simplest thing is to use
 `add_halide_library` and `add_halide_python_extension_library()`:
 
 ```
 # Build a Halide library as you usually would, but be sure to include `PYTHON_EXTENSION`
-add_halide_library(my_filter
-                   GENERATOR my_filter_generator
-                   SOURCES my_filter_generator.cpp
+add_halide_library(xor_filter
+                   GENERATOR logical_op_generator
+                   SOURCES logical_op_generator.py
```

Comment:
I'm confused by this... shouldn't this just be `xor_filter FROM logical_op_generator.py`?

Reply:
Yep, fixed